### ISSUE TITLE
Installation issue: fltk

### ISSUE THREAD

**Issue by antiskid56** [LLNL/spack#22095]
### Steps to reproduce the issue
$ spack install octave@6.2.0+fftw+fontconfig+freetype+gnuplot+curl+fltk
$ [path-to-octave]/bin/octave-6.2.0
octave:1> graphics_toolkit
ans = fltk
octave:2> sombrero 
[error message includes "libfltk_gl.so.1.3: undefined symbol: XGetUtf8FontAndGlyph"]
octave:3> 

```console
$ spack install octave@6.2.0+fftw+fontconfig+freetype+gnuplot+curl+fltk
```

### Information on your system

* **Spack:** 0.16.1-1571-0c38c86403
* **Python:** 3.6.9
* **Platform:** linux-ubuntu18.04-haswell
* **Concretizer:** original

### Additional information

It is not a build-time failure but a run-time failure

None exist.
### General information

- [ x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [ x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [ x] I have uploaded the build log and environment files
- [ ] I have searched the issues of this repo and believe this is not a duplicate

**Comment by antiskid56** [comment]
This is related to `Installation issue: octave #21650`, and `octave: provides fltk with opengl, adds qscintilla #21903`

The patch #21903 to octave allows linking properly to fltk@1.3.3, but when attempting to plot, I get a run time error as described above.

Based on information found in web searches, I tried adding --enable-xft to the configuration for fltk@1.3.3, and adding libxft as a dependency. However this was not enough to clear the runtime error. It appears likely that some additional patch to the fltk@1.3.3 code would be required to deal with this run-time issue.

Rather than attempt another patch to fltk 1.3.3, I propose that the additional version fltk 1.3.4-2 be made available for installation with spack; I have found that this version produces proper plots with octave 6.2.0, without any patching necessary.

I am providing below a proposed patch for the fltk package.py file, which will make fltk version 1.3.4-2 available for installation, but retain version 1.3.3 as the preferred version. Note that disabling xft support is just making explicit what was already implicit in that the package did not incorporate libxft as a dependency.

Likely it would be useful to have a working +xft variant available for this package; however I haven't been able to figure out how to make that work properly yet. With 1.3.3 it builds but has a run-time error; with 1.3.4-2 it doesn't build. Likely it could build with some additional configuration logic to help it find the libxft header files. This is not necessary to get octave plotting working however, so I have not addressed it in this proposed patch.

```console
diff --git a/var/spack/repos/builtin/packages/fltk/package.py b/var/spack/repos/builtin/packages/fltk/package.py
index 2c1f93578d..e9e1fe44a7 100644
--- a/var/spack/repos/builtin/packages/fltk/package.py
+++ b/var/spack/repos/builtin/packages/fltk/package.py
@@ -19,14 +19,19 @@ class Fltk(Package):
 
     """
     homepage = 'http://www.fltk.org/'
-    url = 'http://fltk.org/pub/fltk/1.3.3/fltk-1.3.3-source.tar.gz'
+    def url_for_version(self, version):
+        url = 'http://fltk.org/pub/fltk/{0}/fltk-{1}-source.tar.gz'
+        return url.format(version.up_to(3), version)
 
-    version('1.3.3', sha256='f8398d98d7221d40e77bc7b19e761adaf2f1ef8bb0c30eceb7beb4f2273d0d97')
+    version('1.3.4-2',
+            sha256='25d349c18c99508737d48f225a2eb26a43338f9247551cab72a317fa42cda910')
+    version('1.3.3',
+            sha256='f8398d98d7221d40e77bc7b19e761adaf2f1ef8bb0c30eceb7beb4f2273d0d97',
+            preferred=True)
 
     depends_on('libx11')
 
     patch('font.patch', when='@1.3.3')
-
     # https://github.com/fltk/fltk/commits/master/src/Fl_Tree_Item.cxx
     #  -Fix return value test, as pointed out by Albrecht.
     patch('fix_compare_val.patch', when='@:1.3.3')
@@ -42,7 +47,8 @@ def install(self, spec, prefix):
         options = ['--prefix=%s' % prefix,
                    '--enable-localjpeg',
                    '--enable-localpng',
-                   '--enable-localzlib']
+                   '--enable-localzlib',
+                   '--disable-xft']
 
         if '+shared' in spec:
             options.append('--enable-shared')
```

**Comment by cessenat** [comment]
I've checked that the issue is solved since @sethrj merged PR #26423
This issue should now be closed

$ spack install --jobs 2 octave@6.2.0+fftw+fontconfig+freetype+gnuplot+curl+fltk
```
==> Installing octave-6.2.0-o5j3hcytij7sm2okpuc5yaibd4djgx2f
==> No binary for octave-6.2.0-o5j3hcytij7sm2okpuc5yaibd4djgx2f found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/45/457d1fda8634a839e2fd7cfc55b98bd56f36b6ae73d31bb9df43dde3012caa7c.tar.gz
==> Ran patch() for octave
==> octave: Executing phase: 'autoreconf'
==> octave: Executing phase: 'configure'
==> octave: Executing phase: 'build'
==> octave: Executing phase: 'install'
==> octave: Successfully installed octave-6.2.0-o5j3hcytij7sm2okpuc5yaibd4djgx2f
  Fetch: 14.00s.  Build: 15m 13.11s.  Total: 15m 27.11s.
[+] /home2/cessenat/Softs/spack/opt/spack/linux-ubuntu18.04-zen/gcc-7.5.0/octave-6.2.0-o5j3hcytij7sm2okpuc5yaibd4djgx2f
cessenat@jliana:/home2/cessenat/Softs/spack$ spack load /o5j3hcytij7sm2okpuc5yaibd4djgx2f
cessenat@jliana:/home2/cessenat/Softs/spack$ octave
GNU Octave, version 6.2.0
Copyright (C) 2021 The Octave Project Developers.
This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  For details, type 'warranty'.

Octave was configured for "x86_64-pc-linux-gnu".

Additional information about Octave is available at https://www.octave.org.

Please contribute if you find this software useful.
For more information, visit https://www.octave.org/get-involved.html

Read https://www.octave.org/bugs.html to learn how to submit bug reports.
For information about changes from previous versions, type 'news'.

octave:1> graphics_toolkit
ans = fltk
octave:2> sombrero
octave:3> quit
```
sombrero did plot a nice colored surface in 3d as exepcted

$ spack debug report
* **Spack:** 0.16.2-4625-d9f25f223f
* **Python:** 3.6.9
* **Platform:** linux-ubuntu18.04-zen2
* **Concretizer:** clingo

**Comment by cessenat** [comment]
Oh, I must apologize, it already worked right **before** that PR ; anyway still works thereafter.

$ spack debug report
* **Spack:** 0.16.2-4645-d603f4519d
* **Python:** 3.6.9
* **Platform:** linux-ubuntu18.04-zen2
* **Concretizer:** clingo

**Comment by sethrj** [comment]
Closed by #26423
